pkg/apis/kops/cluster.go

```diff
@@ -400,6 +400,16 @@ type KubeDNSConfig struct {
 	CPURequest *resource.Quantity `json:"cpuRequest,omitempty"`
 	// MemoryLimit specifies the memory limit of each dns container in the cluster. Default 170m.
 	MemoryLimit *resource.Quantity `json:"memoryLimit,omitempty"`
+	// NodeLocalDNS specifies the configuration for the node-local-dns addon
+	NodeLocalDNS *NodeLocalDNSConfig `json:"nodeLocalDNS,omitempty"`
+}
+
+// NodeLocalDNSConfig are options of the node-local-dns
+type NodeLocalDNSConfig struct {
+	// Disable indicates we do not wish to run the node-local-dns addon
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
I think we actually should make this `Enabled *bool`, because we want to differentiate between:

```
nodeLocalDNSConfig:
  enabled: false
  localIP: 10.0.0.1
```

and 

```
nodeLocalDNSConfig:
  localIP: 10.0.0.1
```

Reply:
I suppose this could be a change we do if we consider enabling this by default. If not, we don't really need to distinguish between the two now.